AMBuildScript

```diff
@@ -569,12 +679,29 @@ Extension.configure()
 if Extension.use_auto_versioning():
   Extension.generated_headers = builder.Build('buildbot/Versioning')
 
+if os.path.exists(os.path.join(Extension.sm_root, 'public', 'safetyhook')):
+    # we need to pull safetyhook in locally because ambuild does not take kindly to outside relpaths
+    safetyhook_dest = Normalize(builder.sourcePath + '/safetyhook/')
+    shutil.copytree(os.path.join(Extension.sm_root, 'public', 'safetyhook'), safetyhook_dest, dirs_exist_ok=True)
+
+    SafetyHook = SafetyHookShim()
+    SafetyHook.all_targets = Extension.all_targets
+    builder.Build('safetyhook/AMBuilder', {'SafetyHook': SafetyHook })
+    Extension.libsafetyhook = SafetyHook.libsafetyhook
```

Comment:
I think it would make better sense to have safetyhook as a submodule of this repository, rather than copying safetyhook. But this is fine, feel free to leave as is.

Reply:
yeah i think i can keep it like this